### PR TITLE
[Doc]-Precise topic publish rate limiting

### PR DESCRIPTION
**Pull request by Huanli-Meng**
### Motivation
This doc PR is updated for configurations for PRs:
https://github.com/apache/pulsar/pull/6716
https://github.com/apache/pulsar/pull/6853
https://github.com/apache/pulsar/pull/6074

1: The broker configuration (for https://github.com/apache/pulsar/pull/6716) is updated by Jia Zhai.

2: Add other supported configurations to the client, standlone and proxy configuration docs based on the client.config, standlone.config and proxy.config files.

### Modifications
1: Add TLS with keystore type config in standlone and proxy configuration file.
2: update reference > pulsar configuration > client for PIP-55: Refresh Authentication Credentials
Add other supported configurations to the standlone and proxy configuration files based on the standlone.config and proxy.config files.

### CODE REVIEW

**Comment by Huanli-Meng** [comment]
@jiazhai, please help review the doc update.

**Comment by codelipenghui** [comment]
@Huanli-Meng Could you please rebase the branch?

File: site2/docs/reference-configuration.md [hunk 1]
```diff
@@ -362,7 +369,10 @@ The [`pulsar-client`](reference-cli-tools.md#pulsar-client) CLI tool can be used
 |webServicePort|  THe port used by the standalone broker for HTTP requests  |8080|
 |bindAddress| The hostname or IP address on which the standalone service binds  |0.0.0.0|
 |advertisedAddress| The hostname or IP address that the standalone service advertises to the outside world. If not set, the value of `InetAddress.getLocalHost().getHostName()` is used.  ||
+| numIOThreads | Number of threads to use for Netty IO | 2 * Runtime.getRuntime().availableProcessors() |
+| numHttpServerThreads | Number of threads to use for HTTP requests processing | 2 * Runtime.getRuntime().availableProcessors()|
 |clusterName| The name of the cluster that this broker belongs to. |standalone|
+| failureDomainsEnabled | nable cluster's failure-domain which can distribute brokers into logical region. | false |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:375]:
```suggestion
| failureDomainsEnabled | Enable cluster's failure-domain which can distribute brokers into logical region. | false |
```

File: site2/docs/reference-configuration.md [hunk 2]
```diff
@@ -372,21 +382,86 @@ The [`pulsar-client`](reference-cli-tools.md#pulsar-client) CLI tool can be used
 |ttlDurationDefaultInSeconds|  The default ttl for namespaces if ttl is not configured at namespace policies.  |0|
 |brokerDeleteInactiveTopicsEnabled| Enable the deletion of inactive topics. |true|
 |brokerDeleteInactiveTopicsFrequencySeconds|  How often to check for inactive topics, in seconds. |60|
+| maxPendingPublishdRequestsPerConnection | Maximum pending publish requests per connection to avoid keeping large number of pending requests in memory | 1000|
 |messageExpiryCheckIntervalInMinutes| How often to proactively check and purged expired messages. |5|
 |activeConsumerFailoverDelayTimeMillis| How long to delay rewinding cursor and dispatching messages when active consumer is changed.  |1000|
+| subscriptionExpirationTimeMinutes | How long to delete inactive subscriptions from last consuming. When it is set to 0, inactive subscriptions are not deleted automatically | 0 |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:388]:
```suggestion
| subscriptionExpirationTimeMinutes | How long to delete inactive subscriptions from last consumption. When it is set to 0, inactive subscriptions are not deleted automatically | 0 |
```

File: site2/docs/reference-configuration.md [hunk 3]
```diff
@@ -372,21 +382,86 @@ The [`pulsar-client`](reference-cli-tools.md#pulsar-client) CLI tool can be used
...
+| subscriptionRedeliveryTrackerEnabled | Enable subscription message redelivery tracker to send redelivery count to consumer. | true |
+| subscriptionKeySharedUseConsistentHashing | In the Key_Shared subscription mode, with default AUTO_SPLIT mode, use splitting ranges or consistent hashing to reassign keys to new consumers. | false |
+| subscriptionKeySharedConsistentHashingReplicaPoints | In the Key_Shared subscription mode, number of points in the consistent-hashing ring. The greater the number, the more equal the assignment of keys to consumers. | 100 |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:391]:
```suggestion
| subscriptionKeySharedConsistentHashingReplicaPoints | In the Key_Shared subscription mode, the number of points in the consistent-hashing ring. The greater the number, the more equal the assignment of keys to consumers. | 100 |
```

File: site2/docs/reference-configuration.md [hunk 4]
```diff
@@ -372,21 +382,86 @@ The [`pulsar-client`](reference-cli-tools.md#pulsar-client) CLI tool can be used
...
+| subscriptionExpiryCheckIntervalInMinutes | How frequently to proactively check and purge expired subscription |5 |
+| brokerDeduplicationEnabled | Set the default behavior for message deduplication in the broker. This can be overridden per-namespace. If it is enabled, the broker rejects messages that are already stored in the topic. | false |
+| brokerDeduplicationMaxNumberOfProducers | Maximum number of producer information that it's going to be persisted for deduplication purposes | 10000 |
+| brokerDeduplicationEntriesInterval | Number of entries after which a deduplication information snapshot is taken. A greater interval leads to less snapshots being taken though it would increase the topic recovery time, when the entries published after the snapshot need to be replayed. | 1000 |
+| brokerDeduplicationProducerInactivityTimeoutMinutes | The time of inactivity (in minutes) after which the broker discards deduplication information related to a disconnected producer. | 360 |
+| defaultNumberOfNamespaceBundles | When a namespace is created without specifying the number of bundle, this value is used as the default setting.| 4 |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:397]:
```suggestion
| defaultNumberOfNamespaceBundles | When a namespace is created without specifying the number of bundles, this value is used as the default setting.| 4 |
```

File: site2/docs/reference-configuration.md [hunk 5]
```diff
@@ -372,21 +382,86 @@ The [`pulsar-client`](reference-cli-tools.md#pulsar-client) CLI tool can be used
...
 |statusFilePath|  The path for the file used to determine the rotation status for the broker when responding to service discovery health checks |/usr/local/apache/htdocs|
 |maxUnackedMessagesPerConsumer| The maximum number of unacknowledged messages allowed to be received by consumers on a shared subscription. The broker will stop sending messages to a consumer once this limit is reached or until the consumer begins acknowledging messages. A value of 0 disables the unacked message limit check and thus allows consumers to receive messages without any restrictions. |50000|
 |maxUnackedMessagesPerSubscription| The same as above, except per subscription rather than per consumer.  |200000|
+| maxUnackedMessagesPerBroker | Maximum number of unacknowledged messages allowed per broker. Once this limit reaches, the broker stops dispatching messages to all shared subscription which has higher number of unacknowledged messages until subscriptions start acknowledging messages back and unacknowledged messages count reaches to limit/2. When the value is set to 0, unacknowledged message limit check is disabled and broker does not block dispatchers. | 0 |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:403]:
```suggestion
| maxUnackedMessagesPerBroker | Maximum number of unacknowledged messages allowed per broker. Once this limit reaches, the broker stops dispatching messages to all shared subscriptions which has a higher number of unacknowledged messages until subscriptions start acknowledging messages back and unacknowledged messages count reaches to limit/2. When the value is set to 0, unacknowledged message limit check is disabled and broker does not block dispatchers. | 0 |
```

File: site2/docs/reference-configuration.md [hunk 6]
```diff
@@ -372,21 +382,86 @@ The [`pulsar-client`](reference-cli-tools.md#pulsar-client) CLI tool can be used
...
+| maxUnackedMessagesPerSubscriptionOnBrokerBlocked | Once the broker reaches maxUnackedMessagesPerBroker limit, it blocks subscriptions which has higher unacknowledged messages than this percentage limit and subscription does not receive any new messages until that subscription acknowledges messages back. | 0.16 |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:404]:
```suggestion
| maxUnackedMessagesPerSubscriptionOnBrokerBlocked | Once the broker reaches maxUnackedMessagesPerBroker limit, it blocks subscriptions which have higher unacknowledged messages than this percentage limit and subscription does not receive any new messages until that subscription acknowledges messages back. | 0.16 |
```

File: site2/docs/reference-configuration.md [hunk 7]
```diff
@@ -372,21 +382,86 @@ The [`pulsar-client`](reference-cli-tools.md#pulsar-client) CLI tool can be used
...
 |maxNumPartitionsPerPartitionedTopic|Max number of partitions per partitioned topic. Use 0 or negative number to disable the check|0|
+| topicPublisherThrottlingTickTimeMillis | Tick time to schedule task that checks topic publish rate limiting across all topics. A lower value can give more accuracy while throttling publish but it uses more CPU to perform frequent check. When the value is set to 0, publish throttling is disabled. | 2|
+| brokerPublisherThrottlingTickTimeMillis | Tick time to schedule task that checks broker publish rate limiting across all topics. A lower value can give more accuracy while throttling publish but it uses more CPU to perform frequent check. When the value is set to 0, publish throttling is disabled. |50 |
+| brokerPublisherThrottlingMaxMessageRate | Maximum rate (in 1 seconds) of messages allowed to publish for a broker if the message rate limiting is enabled. When the value is set to 0, message rate limiting is disabled. | 0|
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:408]:
```suggestion
| brokerPublisherThrottlingMaxMessageRate | Maximum rate (in 1 second) of messages allowed to publish for a broker if the message rate limiting is enabled. When the value is set to 0, message rate limiting is disabled. | 0|
```

File: site2/docs/reference-configuration.md [hunk 8]
```diff
@@ -372,21 +382,86 @@ The [`pulsar-client`](reference-cli-tools.md#pulsar-client) CLI tool can be used
...
+| brokerPublisherThrottlingMaxByteRate | Maximum rate (in 1 seconds) of bytes allowed to publish for a broker if the  byte rate limiting is enabled. When the value is set to 0, byte rate limiting is disabled. | 0 |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:409]:
```suggestion
| brokerPublisherThrottlingMaxByteRate | Maximum rate (in 1 second) of bytes allowed to publish for a broker if the  byte rate limiting is enabled. When the value is set to 0, the byte rate limiting is disabled. | 0 |
```

File: site2/docs/reference-configuration.md [hunk 9]
```diff
@@ -403,9 +478,23 @@ The [`pulsar-client`](reference-cli-tools.md#pulsar-client) CLI tool can be used
...
+| bookkeeperTLSClientAuthentication | Enable TLS authentication with bookie. | false |
+| bookkeeperTLSKeyFileType | Supported type: PEM, JKS, PKCS12.  | PEM |
+| bookkeeperTLSTrustCertTypes | Supported type: PEM, JKS, PKCS12.  | PEM |
+| bookkeeperTLSKeyStorePasswordPath | Path to file containing keystore password, if the client keystore is password protected. | | bookkeeperTLSTrustStorePasswordPath | Path to file containing truststore password, if the client truststore is password protected. | |
+| bookkeeperTLSKeyFilePath | Path for the TLS private key file. | |
+| bookkeeperTLSCertificateFilePath | Path for the TLS certificate file. | |
+| bookkeeperTLSTrustCertsFilePath | Path for the trusted TLS certificate file. | |
+| bookkeeperDiskWeightBasedPlacementEnabled | Enable/Disable disk weight based placement. | false |
+| bookkeeperExplicitLacIntervalInMills | Set the interval to check the need for sending an explicit LAC. When the value is set to 0, no explicit LAC is sent. | 0 |
+| bookkeeperClientExposeStatsToPrometheus | Expose BookKeeper client managed ledger stats to prometheus. | false |
```
Comment (Anonymitaet) [inline site2/docs/reference-configuration.md:491]:
```suggestion
| bookkeeperClientExposeStatsToPrometheus | Expose BookKeeper client managed ledger stats to Prometheus. | false |
```

**Comment by Huanli-Meng** [comment]
close the PR and re-open a new one.

**Pull request closed without merging**